File: sdk/python/cmd/pulumi-language-python/testdata/toml/local/projects/l3-component-provider-inheritance/local.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_component as component

class Local(pulumi.ComponentResource):
    def __init__(self, name: str, opts: Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:Local", name, {}, opts)

        # No provider options here: the providers map must be inherited from the
        # enclosing local component and flow through the remote component's
        # registration into its construct call.
        mlc = component.ComponentForeignChild(f"{name}-mlc", value=True,
        opts = pulumi.ResourceOptions(parent=self))

        self.result = mlc.value
        self.register_outputs({
            'result': mlc.value
        })

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/tsnode/projects/l3-component-provider-inheritance/local.ts
import * as pulumi from "@pulumi/pulumi";
import * as component from "@pulumi/component";

export class Local extends pulumi.ComponentResource {
    public result: pulumi.Output<boolean>;
    constructor(name: string, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:Local", name, {}, opts);
        // No provider options here: the providers map must be inherited from the
        // enclosing local component and flow through the remote component's
        // registration into its construct call.
        const mlc = new component.ComponentForeignChild(`${name}-mlc`, {value: true}, {
            parent: this,
        });

        this.result = mlc.value;
        this.registerOutputs({
            result: mlc.value,
        });
    }
}
